packages/lesswrong/lib/editor/make_editable.js

```diff
@@ -0,0 +1,153 @@
+const defaultOptions = {
+  // Determines whether to use the comment editor configuration (e.g. Toolbars)
+  commentEditor: false,
+  // Determines whether to use the comment editor styles (e.g. Fonts)
+  commentStyles: false,
+  // Determines whether to use the comment local storage restoration system
+  commentLocalStorage: false,
+  permissions: {
+    viewableBy: ['guests'],
+    editableBy: ['members'],
+    insertableBy: ['members']
+  },
+  fieldName: "",
+}
+
+export const makeEditable = ({collection, options = {}}) => {
+  options = {...defaultOptions, ...options}
+  const {
+    commentEditor,
+    commentStyles,
+    commentLocalStorage,
+    formGroup,
+    adminFormGroup,
+    permissions,
+    fieldName
+  } = options
+
+  collection.addField([
+    /**
+      Draft-js content
+    */
+    {
+      fieldName: `${fieldName}content`,
+      fieldSchema: {
+        type: Object,
+        optional: true,
+        ...permissions,
+        control: 'EditorFormComponent',
+        blackbox: true,
+        group: formGroup,
+        form: {
+          hintText:"Plain Markdown Editor",
+          multiLine:true,
+          fullWidth:true,
+          disableUnderline:true,
+          commentEditor,
+          commentStyles,
+          commentLocalStorage
+        },
+      }
+    },
+
+    /**
+      Html Body field, made editable to allow access in edit form
+    */
+    {
+      fieldName: `${fieldName}htmlBody`,
+      fieldSchema: {
+        type: String,
+        optional: true,
+        viewableBy: ['guests'],
+        editableBy: ['admins'],
+        insertableBy: ['admins'],
+        control: "textarea",
+        group: adminFormGroup,
+      }
+    },
+
+    /*
+      body: Stores a markdown version of the post
+    */
+
+    {
+      fieldName: `${fieldName}body`,
+      fieldSchema: {
+        type: String,
+        viewableBy: ['guests'],
+        insertableBy: ['members'],
+        editableBy: ['members'],
+        control: "textarea",
+        optional: true,
+        hidden: true,
+        max: 1000000, //overwriting Vulcan's character limit
+      }
+    },
+
+    /*
+      htmlHighlight: stores an HTML highlight of an HTML body
+    */
+
+    {
+      fieldName: `${fieldName}htmlHighlight`,
```

Comment:
Are we just adding this to comments (and presumably not using it?)

Reply:
Yep, we are just adding that stuff to comments, not using it. Was less complexity than adding a bunch of optional parameters, and doesn't hurt anyone.